### PR TITLE
feat: implement storybook of all modal components

### PR DESCRIPTION
**Pull request by Dorvin**
- implement storybook of all modal components
- decompose modal into component and container if needed

### CODE REVIEW

File: components/modal/withdraw-modal.tsx [hunk 1]
```diff
@@ -1,67 +1,41 @@
-import React, { useMemo, useState } from 'react'
-import { formatUnits, isAddressEqual, parseUnits } from 'viem'
-import { useQuery } from 'wagmi'
+import React from 'react'
+import { formatUnits } from 'viem'
 import { min } from 'hardhat/internal/util/bigint'
```
Comment (Dorvin) [inline components/modal/withdraw-modal.tsx:3]: p3; `hardhat bigint min` seems to be compatible with storybook

File: components/modal/repay-modal.tsx [hunk 2]
```diff
@@ -1,153 +1,73 @@
...
+    swapData: `0x${string}`,
+  ) => Promise<void>
+  repay: (
+    position: LoanPosition,
+    amount: bigint,
+    expectedProceeds: bigint,
+  ) => Promise<void>
+  amount: bigint
+  refund: bigint
+  minBalance: bigint
```
Comment (graykode) [inline components/modal/repay-modal.tsx:69]: How about this?
```suggestion
  availableBalance: bigint
```

File: components/modal/withdraw-modal.tsx (hunk 1 above)
Comment (graykode) [reply components/modal/withdraw-modal.tsx:3]: p2; How about pass `disable` parameter from Container?

**Comment by graykode** [comment]
1. I believe it's better to pass `disable` and button text from the `Container`. Additionally, we can remove the hardhat dependency for the `min` function.
2. Please consider adding an `Action` postfix for transaction context!

**Comment by graykode** [comment]
How about creating a `ModalType` to handle overlapping properties? For example:

- position
- onClose
- value
- setValue
- prices
- balances
... and so on

**Review by graykode — APPROVED** [review]

**Pull request merged**